### ISSUE TITLE
Doesn't work at all

### ISSUE THREAD

**Issue by adefaria** [sneat/patreon-downloader#12]
Doesn't work at all

**Comment by sneat** [comment]
Hi @adefaria I'll need some more context.

1. Are you trying to use it on an individual post page?
2. Is it a public post? And if it isn't, are you logged in and do you have access to the post contents?

**Comment by adefaria** [comment]
I guess I wasn't. I expected the download button to either work or complain. It did neither. Nor did the cancel button. Looked like the extension did nothing.

No, it's not a public post, but I was logged in and I had access to the content. 

Perhaps I don't understand what this plugin is supposed to do. At an individual post, it says it scrapes the page. The result is it makes a directory (/tmp/Patreon_Downloader/subfoler/<pagename>) and captures the image of the video I wish to download. It does not download the video, which is what I'm trying to accomplish.

If I try it at the feed level (many posts) it scrapes and moves the page down, seemingly taking a snapshot of every individual post video image but again it does not download the videos themselves.

Thinking perhaps that the video must be playing I play the video and then try Pateron downloader and it gets no images, no video.

How is this supposed to work WRT downloading videos? I want to download the videos for my personal use. I am a paying Patreaon subscriber to this channel.

**Comment by sneat** [comment]
I feel like you're talking about different software @adefaria . This extension doesn't move the page around and doesn't support the feed level at all. Perhaps you're referring to a different extensions?

However, if you're looking to download videos from Patreon, I've had success with this extension in the past: https://chrome.google.com/webstore/detail/live-stream-downloader/looepbdllpjgdmkpdcdffhdbmpbcfekj

**Comment by adefaria** [comment]
Nope. It's this extension and it behaves as I've described it. Here's a video - https://cloud.defariahome.com/index.php/s/p6e9inXa88tmKAL.

And here's another video showing problems with your Live Stream Downloader - https://cloud.defariahome.com/index.php/s/fnC2GR6gPZXjmxY

**Comment by sneat** [comment]
Thanks for the videos @adefaria it certainly cleared up what's happening for you.

I maintain [this patreon downloader](https://chrome.google.com/webstore/detail/patreon-downloader/mnfjhjpninhcccbahcdbcphpifofoajc) extension which is not the one you are using. You're using [this one](https://chrome.google.com/webstore/detail/patreon-downloader/fmhpbbdpdhjkpbklloabkncldkmggaik) (which has the same name), which at least explains the confusion on my side.

That said though, my one won't help you either. Downloading videos from patreon is a painful process at the best of times, let alone when they are vimeo videos (as you noticed, they are embedded within iframes - 2 in fact).

For vimeo in particular, this [one specifically for vimeo](https://chrome.google.com/webstore/detail/video-downloader-for-vime/cgmcdpfpkoildicgacgldinemhgmcbgp/) might work (you'll likely need to click play on the video before it shows up at all) 🤞 

I'm not going to be able to add video support into this extension, so if the above `Video Downloader for Vimeo` doesn't work, my recommendation will be to keep trying various other video downloader extensions and hopefully one of them works. Please let me know which one ends up working for you.

**Comment by adefaria** [comment]
Even your extension page says it will download all media. Aren't videos media? Or are you saying you can download videos but vimeo videos encased in a few iframes is not supported.

The other extension you pointed me to didn't work either. Just said error.

I used to use this site - https://youtube4kdownloader.com/en58/, which required using the iframe URL but then scraped the page and presented the video download links of the various resolutions. Now clicking on the download links just errors out with

"An error occurred from remote video site. Please try with other download link".

**Comment by sneat** [comment]
I don't really understand why you're arguing with me about whether my extension will download videos. It downloads the files that patreon makes available in its data. Patreon actively make it hard to download videos from their site and are constantly changing their interface to ensure it's difficult. The extension itself says that it probably won't be able to download any videos. I didn't update the extension description because for some creators, it still can download videos. It entirely depends on how the creator is adding their videos.

Vimeo makes downloading videos hard as well. You might have luck by clicking on the video and getting it to open on vimeo's website but you also might not.

I've tried to give you options even when your original request wasn't actually about my extension (I understand the naming confusion), but I really can't help any more than that.

The extension I made doesn't support video embeds from vimeo, so unless your creator starts posting videos in a different way, you'll need to find another option.
